src/ctapipe/tools/calculate_pixel_stats.py

```diff
@@ -93,24 +93,35 @@ def setup(self):
                 parent=self,
             )
         )
-        # Check that the input and output files are not the same
-        if self.input_data.input_url == self.output_path:
-            raise ToolConfigurationError(
-                "Input and output files are same. Fix your configuration / cli arguments."
-            )
+        # Check that the DL1 images are selected in the TableLoader
         if "dl1_images" in self.input_data.config.TableLoader:
```

Comment:
I'm not sure this logic is correct - the config only includes keys that the user has specified on the command-line or config file, and the default value for `dl1_images` is False.  So this would only work if the user ran with the explicit option `--TableLoader.dl1_images=False`

You don't really need to check the config, you can just check if `if not self.input_data.dl1_images`, since that will be set after the TableLoader is constructed to the default or to the value the user specified.

Reply:
this is a way to override the default in case it is not explicitly set

Reply:
But now you force it anyway after. Please don't "butcher" the config system in this way. We know it's not really great to work with, but we should minimize these workarounds as much as possible. They tend to break.

Reply:
Not really, you can still issue a warning by comparing `self.input_data.config.TableLoader.dl1_images` and self.input_data.dl1_images` if you want. But it might not always work, and certainly will break when we change the config system

Reply:
By the way, it's not in the standard repr, but the HTML in a notebook one shows you explicitly which parameters have been changed from the defaults.  Not really useful for a command-line-tool, but worth mentioning.  You can also see what was overridden and used in the provenance outptu, which contains both the input config and the final config that was used after construction. So technically the user can see what is happening to their options, even if a bit convoluted